runners/direct-java/src/main/java/org/apache/beam/runners/direct/DirectRunner.java

```diff
@@ -179,6 +180,11 @@ public DirectPipelineResult run(Pipeline pipeline) {
 
       DisplayDataValidator.validatePipeline(pipeline);
       DisplayDataValidator.validateOptions(options);
+      // TODO(BEAM-10670): Remove the deprecated Read and make the splittable DoFn the only option.
+      if (!(ExperimentalOptions.hasExperiment(options, "beam_fn_api_use_deprecated_read")
```

Comment:
I wanted to use one that wasn't coupled to `beam_fn_api` so that users of non portable pipelines wouldn't be confused. I also don't want to impact existing users until we start removing the experiment so I needed to support both.